crates/nu-parser/src/parser.rs

```diff
@@ -2449,6 +2453,94 @@ pub fn unescape_string(bytes: &[u8], span: Span) -> (Vec<u8>, Option<ParseError>
+                Some(b'x') => {
+                    match (bytes.get(idx + 1), bytes.get(idx + 2)) {
+                        (Some(h1), Some(h2)) => {
+                            let s = String::from_utf8(vec![*h1, *h2]);
+
+                            if let Ok(s) = s {
+                                let int = u32::from_str_radix(&s, 16);
+
+                                if let Ok(int) = int {
+                                    let result = char::from_u32(int);
+
+                                    if let Some(result) = result {
+                                        let mut buffer = vec![0; 4];
+                                        let result = result.encode_utf8(&mut buffer);
+
+                                        for elem in result.bytes() {
+                                            output.push(elem);
+                                        }
+
+                                        idx += 3;
+                                        continue;
+                                    }
+                                }
+                            }
+                            err = Some(ParseError::Expected(
+                                "hex value".into(),
+                                Span {
+                                    start: (span.start + idx),
+                                    end: span.end,
+                                },
+                            ));
+                        }
+                        _ => {
+                            err = Some(ParseError::Expected(
+                                "hex value".into(),
+                                Span {
+                                    start: (span.start + idx),
+                                    end: span.end,
+                                },
+                            ));
+                        }
+                    }
+                    idx += 3;
+                }
+                Some(b'o') => {
+                    match (bytes.get(idx + 1), bytes.get(idx + 2), bytes.get(idx + 3)) {
+                        (Some(h1), Some(h2), Some(h3)) => {
+                            let s = String::from_utf8(vec![*h1, *h2, *h3]);
+
+                            if let Ok(s) = s {
+                                let int = u32::from_str_radix(&s, 8);
+
+                                if let Ok(int) = int {
+                                    let result = char::from_u32(int);
+
+                                    if let Some(result) = result {
+                                        let mut buffer = vec![0; 4];
+                                        let result = result.encode_utf8(&mut buffer);
+
+                                        for elem in result.bytes() {
+                                            output.push(elem);
+                                        }
+
+                                        idx += 4;
+                                        continue;
+                                    }
```

Comment:
Same as above

Reply:
I'm backing this out, definitely need more consideration and I am leaning toward saying nushell only needs `\u`